lsst-epo/investigations-client
# [C] Fix login/activation flows

JIRA: https://jira.lsstcorp.org/browse/EPO-8800

## What this change does ##

Fix login and activation flow issues, changes include:

- activation link now works
- user status is updated from pending to active when activation is completed
- change start button to go to first investigation page when authed
- show currently authed user
- show additional message if user is authed but pending
- add create account button when not authed

## Notes for reviewers ##

client side only, no changes to API. 

Include an indication of how detailed a review you want on a 1-10 scale.
- 5 = "Please make sure there are no obvious errors and that you believe it does what it says it does"

## Testing ##

- start logged out
- create a new account
- see that after creation, account is in a pending state without logged in investigation status (user will have to log out or activate to start investigation)
- click activation link and see that activation modal appears, and user is moved to active status
- see that investigation start is now available

## Checklist ##

If any of the following are true, please check them off

## Screenshots (optional) ##
unauthed
![image](https://github.com/lsst-epo/investigations-client/assets/11721838/b0e331e8-877e-4b2e-b18c-03bb79df26fe)

authed but pending
![image](https://github.com/lsst-epo/investigations-client/assets/11721838/10836c03-a909-40be-9d98-4b6a2ff2f78d)

authed
![image](https://github.com/lsst-epo/investigations-client/assets/11721838/0b2a61f8-8cc2-4fb4-8ce6-118280b7f0e9)
